.github/workflows/scan.yml

```diff
@@ -0,0 +1,75 @@
+on:
+  schedule:
+    - cron: '0 1 * * *'
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  sonarcloud: 
+    name: SonarCloud Scan
+    if: ${{ github.event_name != 'schedule' && github.actor != 'dependabot[bot]' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v6
+        with:
+          fetch-depth: 1
```

Comment:
I think this needs to be 0 otherwise sonarcloud complains about a shallow clone (at least that's what was happening for me)